projects/admin/src/app/record/detail-view/item-detail-view/item-transaction/item-transaction.component.ts

```diff
@@ -14,21 +14,211 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-import { Component, Input } from '@angular/core';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { RecordService, DialogService, DialogComponent } from '@rero/ng-core';
+import { ToastrService } from 'ngx-toastr';
+import { LoanState } from 'projects/admin/src/app/circulation/items';
+import { LoanService } from 'projects/admin/src/app/service/loan.service';
+import { UserService } from 'projects/admin/src/app/service/user.service';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 @Component({
   selector: 'admin-item-transaction',
-  templateUrl: './item-transaction.component.html',
-  styles: []
+  templateUrl: './item-transaction.component.html'
 })
-export class ItemTransactionComponent {
-
-  /** Loan Record */
+export class ItemTransactionComponent implements OnInit {
+  /**
+   * Loan Record
+   */
   @Input() transaction: any;
 
-  /** Ressource type */
+  /**
+   * Ressource type
+   */
   @Input() type: string;
 
-  /** Flag for cell background */
+  /**
+   * Flag for cell background
+   */
   @Input() background: boolean;
+
+  /**
+   * Item pid
+   */
+  @Input() itemPid: string;
+
+  /**
+   * Current item loans
+   */
+  loans$: Observable<any>;
+
+  /**
+   * Pickup locations of the organisation
+   */
+  pickupLocations: any;
+
+  /**
+   * Pickup default $ref
+   */
+  private _pickupDefaultValue: string;
+
+  /**
+   * Current user
+   */
+  private _currentUser: any;
+
+  /**
+   * Informs parent component to remove request when it is cancelled
+   */
+  @Output() removeRequest = new EventEmitter<boolean>();
+
+  /**
+   * On init hook
+   */
+  ngOnInit() {
+    this._currentUser = this._userService.getCurrentUser();
+    this.getPickupLocations().subscribe((pickups) => {
```

Comment:
Needs a subscriber to unsubscribe. Check the rest of your code. Sometimes is not required (http request), but for the other subscribe method you have to.